linera-base/src/data_types.rs

```diff
@@ -731,6 +732,77 @@ impl std::str::FromStr for OracleResponse {
     }
 }
 
+/// A blob of binary data.
+#[derive(Eq, PartialEq, Debug, Hash, Clone, Serialize, Deserialize)]
+pub struct Blob {
+    #[serde(with = "serde_bytes")]
+    bytes: Vec<u8>,
+}
+
+impl Blob {
+    /// Loads a hashed blob from a file.
+    pub async fn load_from_file(path: impl AsRef<std::path::Path>) -> std::io::Result<Self> {
+        Ok(Blob {
+            bytes: std::fs::read(path)?,
+        })
+    }
+
+    /// Creates a `HashedBlob` without checking that this is the correct BlobId!
+    pub fn with_hash_unchecked(self, blob_id: BlobId) -> HashedBlob {
+        HashedBlob {
+            id: blob_id,
+            blob: self,
+        }
+    }
+}
+
+impl BcsHashable for Blob {}
+
+impl From<HashedBlob> for Blob {
+    fn from(blob: HashedBlob) -> Blob {
+        blob.blob
+    }
+}
+
+/// A blob of binary data, with it's content-addressed blob ID
+#[derive(Eq, PartialEq, Debug, Hash, Clone)]
+pub struct HashedBlob {
+    id: BlobId,
+    blob: Blob,
+}
+
+impl HashedBlob {
+    /// A content-addressed blob ID i.e. the hash of the Blob.
```

Comment:
```suggestion
    /// A content-addressed blob ID i.e. the hash of the `Blob`.
```

Or "blob" (lowercase).